pkg/tempo/instance.go

```diff
@@ -141,11 +142,19 @@ func (i *Instance) buildAndStartPipeline(ctx context.Context, cfg InstanceConfig
 	}
 
 	if cfg.SpanMetrics != nil && len(cfg.SpanMetrics.PromInstance) != 0 {
-		prom, err := promManager.GetInstance(cfg.SpanMetrics.PromInstance)
-		if err != nil {
-			return fmt.Errorf("failed to get prometheus instance %s", cfg.SpanMetrics.PromInstance)
+		for {
+			select {
+			case <-ctx.Done():
```

Comment:
i don't think this context has a timeout. i think this will spin forever. 

we need to either add a timeout or max retries or something.

Reply:
Nice catch, thanks. I ended up going for @rfratto's suggestion of passing the manager to the exporter.